lib/omnibus/fetchers/net_fetcher.rb

```diff
@@ -125,6 +125,11 @@ def checksum
       source[digest_type]
     end
 
+    # Set to false to extract archives into the root source_dir (default).
+    # Set to true to always extract underneath project_dir
+    #
+    attr_accessor :isolate_extraction
```

Comment:
I guess it depends on what you mean by "backwards compatible".  See https://github.com/chef/omnibus-software/blob/master/config/software/libzmq4x-windows.rb  It assumes that the extracted zip is going to be right under source_dir.  See https://github.com/chef/omnibus/blob/master/lib/omnibus/builder.rb#L355 .  It assumes that the gem you're building is going to be right under source_dir.  I'm sure there are other such implicit assumptions in our CI and in other omnibus definitions.  Now I don't mind breaking that and making people fix their code... but I was wary of fighting too many yaks at once.  If y'all are cool with that, then, indeed, I can make this way simpler.  No need for extract_path.  Every fetcher will always unconditionally extract to a source_dir/package_path and the software DSL will take care of correctly setting project_path to source_dir/package_path/relative_path.